### PR TITLE
Update license classifications from ScanCode licensedb

### LINKED ISSUE #107: Re-import latest ScanCode license classifications
Meanwhile there have been several upstream changes, so we should re-generate the classifications.

1. A new "cla" category has been introduced which makes our hard-coded category mapping at least partially unnecessary.
2. A bunch of new license have been added
3.Several "unknown" licenses have been removed

### PR DESCRIPTION
**Pull request by fviernau**
See individual commits.

Fixes #107.

### CODE REVIEW

**Comment by fviernau** [comment]
@oss-review-toolkit/core-devs:  I just stumbled across the following issue:

1. A particular license ID does not have an SPDX identifier in ScanCode but just a `LicenseRef-*` identifier
2. In a new ScanCode version an Spdx identifier gets added to that license

outcome: ` ./gradlew generateLicenseClassifications` will remove the `LicenseRef-*` identifier in favor of the SPDX one.
This is problematic as one cannot use the `license-classifications.yml` for the older ScanCode version anymore, as the `LicenseRef-*` id would be required there.

Proposal: For each SPDX license add two entries to `license-classifications-yml`: One with SPDX id, one with `LicenseRef-*` id.

File: evaluator-rules/src/main/resources/evaluator.rules.kts [hunk 1]
```diff
@@ -40,7 +40,7 @@ fun getLicensesForCategory(category: String): Set<SpdxSingleLicenseExpression> =
     }
 
 val commercialLicenses = getLicensesForCategory("commercial")
-val contributorLicenseAgreementLicenenses = getLicensesForCategory("contributor-license-agreement")
+val claLicenenses = getLicensesForCategory("cla")
```
Comment (sschuberth) [inline evaluator-rules/src/main/resources/evaluator.rules.kts:43]: Typo in "Licenenses".

**Review by sschuberth — CHANGES_REQUESTED** [review]

**Comment by sschuberth** [comment]
> Proposal: For each SPDX license add two entries to `license-classifications-yml`: One with SPDX id, one with `LicenseRef-*` id.

How would you actually get both the (old) LicenseRef and the (new) SPDX ID for any specific license if the online ScanCode DB only offers the new data?

**Comment by fviernau** [comment]
> How would you actually get both the (old) LicenseRef and the (new) SPDX ID for any specific license if the online ScanCode DB only offers the new data?

It seems to offer both, see e.g.: https://scancode-licensedb.aboutcode.org/fsf-unlimited-no-warranty.json

**Comment by sschuberth** [comment]
> It seems to offer both, see e.g.: https://scancode-licensedb.aboutcode.org/fsf-unlimited-no-warranty.json

Ok, then the proposed approach makes sense to me.

**Review by sschuberth — APPROVED** [review]

**Comment by fviernau** [comment]
> So you would create one entry for the SPDX id and each of the `other_spdx_license_keys`?
> For example Apache-2.0 has more than one: https://scancode-licensedb.aboutcode.org/apache-2.0.json

yes.

**Pull request merged**